### PR TITLE
Search 161

### PR DESCRIPTION
**Pull request by alexbakharew**
### Scope & Purpose

*(Please describe the changes in this PR for reviewers - **mandatory**)*

- [ ] :hankey: Bugfix (requires CHANGELOG entry)
- [X] :pizza: New feature (requires CHANGELOG entry, feature documentation and release notes)
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification
- [ ] :book: CHANGELOG entry made

#### Backports:

- [ ] No backports required
- [x] Backports required for: *(Please specify versions and link PRs)*

#### Related Information

*(Please reference tickets / specification / other PRs etc)*

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [X] GitHub issue / Jira ticket number: Search-161
- [ ] Design document: 

### Testing & Verification

*(Please pick either of the following options)*

- [ ] This change is a trivial rework / code cleanup without any test coverage.
- [ ] The behavior in this PR was *manually tested*
- [ ] This change is already covered by existing tests, such as *(please describe tests)*.
- [ ] This PR adds tests that were used to verify all changes:
  - [X] Added new C++ **Unit tests**
  - [X] Added new **integration tests** (e.g. in shell_server / shell_server_aql)
  - [ ] Added new **resilience tests** (only if the feature is impacted by failovers)
- [ ] There are tests in an external testing repository:
- [ ] I ensured this code runs with ASan / TSan or other static verification tools

Link to Jenkins PR run:

### Documentation

> All new features should be accompanied by corresponding documentation. 
> Bugs and features should furthermore be documented in the CHANGELOG so that
> support, end users, and other developers have a concise overview. 

- [ ] Added entry to *Release Notes* 
- [ ] Added a new section in the *Manual* 
- [ ] Added a new section in the *HTTP API* 
- [ ] Added *Swagger examples* for the HTTP API  
- [ ] Updated license information in *LICENSES-OTHER-COMPONENTS.md* for 3rd party libraries

### External contributors / CLA Note 

Please note that for legal reasons we require you to sign the [Contributor Agreement](https://www.arangodb.com/documents/cla.pdf)
before we can accept your pull requests.

### CODE REVIEW

**Comment by alexbakharew** [comment]
Draft

File: arangod/Aql/AqlFunctionFeature.cpp [hunk 1]
```diff
@@ -195,7 +195,7 @@ void AqlFunctionFeature::addStringFunctions() {
   add({"ENCODE_URI_COMPONENT", ".", flags, &Functions::EncodeURIComponent});
   add({"SOUNDEX", ".", flags, &Functions::Soundex});
   add({"LEVENSHTEIN_DISTANCE", ".,.", flags, &Functions::LevenshteinDistance});
-  add({"LEVENSHTEIN_MATCH", ".,.,.|.,.", flags, &Functions::LevenshteinMatch});  // (attribute, target, max distance, [include transpositions, max terms])
+  add({"LEVENSHTEIN_MATCH", ".,.,.|.,.,.", flags, &Functions::LevenshteinMatch});  // (attribute, target, max distance, [include transpositions, max terms])
```
Comment (Dronplane) [inline arangod/Aql/AqlFunctionFeature.cpp:198]:
```suggestion
  add({"LEVENSHTEIN_MATCH", ".,.,.|.,.,.", flags, &Functions::LevenshteinMatch});  // (attribute, target, max distance, [include transpositions, max terms, prefix])
```

File: tests/IResearch/IResearchFilterFunction-test.cpp [hunk 2]
```diff
@@ -6343,7 +6409,7 @@ TEST_F(IResearchFilterFunctionTest, levenshteinMatch) {
   // wrong number of arguments
   assertFilterParseFail(
       vocbase(),
-      "FOR d IN myView FILTER LEVENSHTEIN_MATCH(d.foo, 'true', 1, false, 1, 'z') RETURN d");
+      "FOR d IN myView FILTER LEVENSHTEIN_MATCH(d.foo, 'true', 1, false, 1, 'z', 'x') RETURN d");
```
Comment (Dronplane) [inline tests/IResearch/IResearchFilterFunction-test.cpp:6412]: Please add a test with invalid parameter type for 'prefix' e,g, array, object or number or null - something like that,

**Review by Dronplane — CHANGES_REQUESTED** [review]
Please add at least one js test to cover new parameter.
Also we have LEVENSHTEIN_MATCH implementation for pure Aql  - should we also support prefix match here ? As for  current implementation - prefix will be accepted  by function (due to changes in signature for function registration) but totally ignored. And that seems weird.

**Review by gnusi — APPROVED** [review]

**Review by Dronplane — APPROVED** [review]
LGTM
